packages/flutter/lib/src/widgets/resizing_header_sliver.dart

```diff
@@ -0,0 +1,213 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'slotted_render_object_widget.dart';
+
+/// A sliver that is pinned to the start of its [CustomScrollView] and
+/// reacts to scrolling by resizing between the intrinsic sizes of its
+/// min and max extent prototypes.
+///
+/// The minimum and maximum sizes of this sliver are defined by [minExtentPrototype]
+/// and [maxExtentPrototype], a pair of widgets that are laid out once. You can
+/// use [SizedBox] widgets to define the size limits.
+///
+/// This sliver is preferable to the general purpose [SliverPersistentHeader]
+/// for its relatively narrow use case because there's no need to create a
+/// [SliverPersistentHeaderDelegate] or to predict the header's minimum or
+/// maximum size.
+///
+/// {@tool dartpad}
+/// This sample shows how this sliver's two extent prototype properties can be used to
+/// create a resizing header whose minimum and maximum sizes match small and large
+/// configurations of the same header widget.
+///
+/// ** See code in examples/api/lib/widgets/sliver/resizing_header_sliver.0.dart **
+/// {@end-tool}
+class ResizingHeaderSliver extends StatelessWidget {
+  /// Create a pinned header sliver that reacts to scrolling by resizing between
+  /// the intrinsic sizes of the min and max extent prototypes.
+  const ResizingHeaderSliver({
+    super.key,
+    this.minExtentPrototype,
+    this.maxExtentPrototype,
+    this.child,
+  });
+
+  /// Laid out once to define the minimum size of this sliver along the
+  /// [CustomScrollView.scrollDirection] axis.
+  ///
+  /// If null, the minimum size of the sliver is 0.
+  final Widget? minExtentPrototype;
+
+  /// Laid out once to define the maximum size of this sliver along the
+  /// [CustomScrollView.scrollDirection] axis.
+  ///
+  /// If null, the maximum size of the sliver is 0.
+  final Widget? maxExtentPrototype;
+
+  /// The widget contained by this sliver.
+  final Widget? child;
+
+  @override
+  Widget build(BuildContext context) {
+    return _ResizingHeaderSliver(
+      minExtentPrototype: minExtentPrototype,
```

Comment:
Still carefully reviewing all the related PRs🙂. Just FTR, here we probably need to use a `ExcludeFocus` to wrap the `minExtentPrototype` and `maxExtentPrototype`; Otherwise, if we put some tappable widgets here, when we do focus traversal by just using `tab` key, these two would also be traversed, even though they are not really painted. Probably we don't need `IgnorePointer` here because hitTest doesn't include them:)

Reply:
Good point, I will fix this stuff and add some tests to check.